### PR TITLE
Add Codex

### PR DESCRIPTION
**Pull request by RaneyDazed**
# Description

Adding Codex, as recently requested. It adds the volume, `/comics` to `/mnt/unionfs/Media/Comics` as currently written. 

# How Has This Been Tested?

Installed via docker compose initially, then installed via saltbox_mod. That is how I'm currently running it. Works fine and I've not gotten any errors so far. the user mapping seems to be working fine, and its still scanning my comic directory. 

- [x] Tested on my feeder, and tested on my aio/mediabox.

### CODE REVIEW

**Comment by RaneyDazed** [comment]
don't approve yet, not sure if I'm having trouble due to not setting up my user in the UI or if its something else. :p works fine via compose, need to dick w the role for a few

**Comment by RaneyDazed** [comment]
Ok, it works as expected now. Pulls up paths and allows you to scan your comics directory.

**Comment by RaneyDazed** [comment]
#236

**Comment by Mariachi2100** [comment]
Awesome.  Thank you for your efforts.  Much appreciated.

**Comment by RaneyDazed** [comment]
> Awesome. Thank you for your efforts. Much appreciated.

Happy to help! Just gotta wait for someone to review it and make sure its ok with the maintainers and it will go live. : )

**Comment by maximuskowalski** [comment]
Did you check if it HAS to be `/comics` for the media dir or can you use whatever you want?

**Comment by RaneyDazed** [comment]
I'll take another quick look. just a sec

**Comment by RaneyDazed** [comment]
I didn't look too deep into it as I'm in bed rn, but this is the compose from their "docker" setup on docker hub:

```yaml
services:
  codex:
      env_file: .env
      image: docker.io/ajslater/codex
      container_name: codex
      volumes:
          - <host path to data>:/config
          - <host path to comics>:/comics:ro
      ports:
          - 9810:9810
      restart: on-failure
```

**Comment by RaneyDazed** [comment]
Screenshot 2023-03-22 at 11 23 44 PM

**Comment by RaneyDazed** [comment]
Might be too late and my brain isn't processing the question correctly. when you ask if it needs to be mapped to `/comics` you mean to ask, do they allow the user to make `/comics` anything they want? because if so, I have no idea. I haven't seen anything that says we can. But I haven't seen anything that says we can't either. I can try to make it whatever tomorrow at work when I should be working : p

**Comment by maximuskowalski** [comment]
Yeh, so I think I would add /mnt:/mnt to this so people can also use whatever library paths they want or might already have. Also need to think about how we add this to the traefik branch at the same time, so don't delete the branch after it gets merged, put in another PR for the traefik3 branch and we can edit it to work there as well.

We should check we are still adding mnt:/mnt and not /mnt/Media for this sort of situation too. There is of course the custom inventory option too.

File: roles/codex/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+################################
+# Paths
+################################
+
+codex_paths_folder: "{{ codex_name }}"
+codex_paths_location: "{{ server_appdata_path }}/{{ codex_paths_folder }}"
+codex_paths_folders_list:
+  - "{{ codex_paths_location }}"
+  - "{{ codex_paths_location }}/config"
+  - "/mnt/unionfs/Media/Comics"
```
Comment (RaneyDazed) [inline roles/codex/defaults/main.yml:25]: Gotta fox this guy.

Comment (RaneyDazed) [reply roles/codex/defaults/main.yml:25]: Remove not fix

File: roles/codex/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+# Commands
+codex_docker_commands_default: []
+codex_docker_commands_custom: []
+codex_docker_commands: "{{ lookup('vars', codex_name + '_docker_commands_default', default=codex_docker_commands_default)
+                           + lookup('vars', codex_name + '_docker_docker_commands_custom', default=codex_docker_commands_custom) }}"
+
+# Volumes
+codex_docker_volumes_default:
+  - "{{ codex_paths_location }}:/config"
+  - "/mnt/unionfs/Media/Comics:/comics"
```
Comment (RaneyDazed) [inline roles/codex/defaults/main.yml:103]: And change this to `/mnt`

File: roles/codex/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+# Commands
+codex_docker_commands_default: []
+codex_docker_commands_custom: []
+codex_docker_commands: "{{ lookup('vars', codex_name + '_docker_commands_default', default=codex_docker_commands_default)
+                           + lookup('vars', codex_name + '_docker_docker_commands_custom', default=codex_docker_commands_custom) }}"
+
+# Volumes
+codex_docker_volumes_default:
+  - "{{ codex_paths_location }}:/config"
+  - "/mnt:/mnt"
```
Comment (saltydk) [inline roles/codex/defaults/main.yml:102]: /mnt is already mounted in all containers, so remove that.

**Comment by RaneyDazed** [comment]
I was scratching my head looking for the default docker volume with `mnt` and couldn't find it. that pesky max came by and did it for me :p tyvm max!

**Comment by maximuskowalski** [comment]
I was actually under the impression that /mnt:/mnt had been removed and was frowned upon being added to all roles so I too was scratching.

**Comment by saltydk** [comment]
We just added an opt-out variable from what I remember

**Comment by RaneyDazed** [comment]
> Yeh, so I think I would add /mnt:/mnt to this so people can also use whatever library paths they want or might already have. Also need to think about how we add this to the traefik branch at the same time, so don't delete the branch after it gets merged, put in another PR for the traefik3 branch and we can edit it to work there as well.
> 
> We should check we are still adding mnt:/mnt and not /mnt/Media for this sort of situation too. There is of course the custom inventory option too.

I think I did this without thinking :p

**Comment by maximuskowalski** [comment]
Are we all good with this one now RD? And we have a docs PR too?

**Comment by RaneyDazed** [comment]
hmm I'll have to see if I've made that already or not. I'll take a look. It shouldn't be too difficult to get done. : ) but yes all set on this one! Aside from the docs PR

> Are we all good with this one now RD? And we have a docs PR too?

**Comment by saltydk** [comment]
@maximuskowalski @owine review this before the end of the weekend, merge it to traefik3 or close it.

**Comment by owine** [comment]
This was merged without a direct test - any issues with the role, please ping me in Discord to troubleshoot

**Pull request merged**